src/lib/profiles/data-management/Current/SubscriptionClient.cpp

```diff
@@ -561,6 +560,16 @@ WEAVE_ERROR SubscriptionClient::SendSubscribeRequest(void)
                 // Start, fill, and close the TLV Structure that contains ResourceID, ProfileID, and InstanceID
                 err = mDataSinkCatalog->HandleToAddress(versionedTraitPath.mTraitDataHandle, writer,
                                                         versionedTraitPath.mRequestedVersionRange);
+
+                if (err == WEAVE_ERROR_INVALID_ARGUMENT)
```

Comment:
If this error is truly unexpected, then this test should really be replaced with a SuccessOrExit().

Reply:
The only reason I saw in keeping it the way it is now is this piece of error check is independent of the previous Locate() error handling. Otherwise, if it were replaced by a SuccessOrExit() and later, for whatever reason, if the earlier Locate() was refactored/moved then the error handling here would break. Not sure what the chances are for that. So, to keep it simple and less code change for now, we could replace it with SuccessOrExit().